src/jit/lclvars.cpp

```diff
@@ -3388,20 +3388,24 @@ void Compiler::lvaMarkLocalVars()
 
 #endif // !FEATURE_EH_FUNCLETS
 
-#if FEATURE_EH_FUNCLETS
-    if (ehNeedsPSPSym())
+    // PSPSym and LocAllocSPvar are not used by the CoreRT ABI
+    if (eeGetEEInfo()->targetAbi != CORINFO_CORERT_ABI)
     {
-        lvaPSPSym            = lvaGrabTempWithImplicitUse(false DEBUGARG("PSPSym"));
-        LclVarDsc* lclPSPSym = &lvaTable[lvaPSPSym];
-        lclPSPSym->lvType    = TYP_I_IMPL;
-    }
+#if FEATURE_EH_FUNCLETS
+        if (ehNeedsPSPSym())
+        {
+            lvaPSPSym            = lvaGrabTempWithImplicitUse(false DEBUGARG("PSPSym"));
+            LclVarDsc* lclPSPSym = &lvaTable[lvaPSPSym];
+            lclPSPSym->lvType    = TYP_I_IMPL;
+        }
 #endif // FEATURE_EH_FUNCLETS
 
-    if (compLocallocUsed)
-    {
-        lvaLocAllocSPvar         = lvaGrabTempWithImplicitUse(false DEBUGARG("LocAllocSPvar"));
-        LclVarDsc* locAllocSPvar = &lvaTable[lvaLocAllocSPvar];
-        locAllocSPvar->lvType    = TYP_I_IMPL;
+        if (compLocallocUsed)
+        {
+            lvaLocAllocSPvar         = lvaGrabTempWithImplicitUse(false DEBUGARG("LocAllocSPvar"));
+            LclVarDsc* locAllocSPvar = &lvaTable[lvaLocAllocSPvar];
+            locAllocSPvar->lvType    = TYP_I_IMPL;
+        }
     }
```

Comment:
I'm *pretty* sure the LocAllocSPvar is only required by x86 (and the implicit frame layout expected by the VM), but I've been too scared there's some other effect, like some implicit debugging dependency, to remove it for other platforms. That's unrelated to this change, of course.